### ISSUE TITLE
xdrip+ > HAPP Feature Request

### ISSUE THREAD

**Issue by kskandispersonal** [timomer/HAPP#47]
Pull xDrip+ treatments (carbs and insulin) into HAPP.

**Comment by jamespaulley** [comment]
Yeah this would be great.

**Comment by timomer** [comment]
Hey, do you both also sync with Nightscout from xDrip+?

If so im thinking its better to sync from NS than xDrip+.

If not, why do you enter this into xDrip+and not just HAPP?

**Comment by jamespaulley** [comment]
I haven't used Nightscout, no.

I like using xDrip+ at the moment, I use it to show carb absorption and insulin action on the graph. I guess it would just be nice if the two apps worked together?

**Comment by diabetesdt1** [comment]
On xDrip+ you can add your IE/KH over the Smartwatch, you dont need to get your mobile phone out from the pocket :-)
And the own NS-Site is blocked/not reachable in my office wlan and in the province, the Internet-Connection is very bad. A local sync/broadcasting is better and stable.

When HAPP works with xDrip+, you can use the Graphs and detailed Settings from xDrip+ and the pump-Settings from HAPP. Then, it would very good

Thanks for your work!

**Comment by timomer** [comment]
👍 Cool, thanks for clarifying!

**Comment by jamespaulley** [comment]
If I enter carb and bolus data in Happ, can I see it in xDrip+?

**Comment by timomer** [comment]
As of now, only if you use Nightscout. I can add this as a feature request for a direct to xDip+ plugin

**Comment by jamespaulley** [comment]
Cool sounds great.

Am I correct in thinking that Nightscout requires a data connection? One of my incentives for using xDrip+ is that it will work when out of cellular data connection.

**Comment by timomer** [comment]
Yep, Nightscout is hosted, see it as more of a place to dump data too and visulise it

**Comment by jamespaulley** [comment]
OK cool I'll check it out, thanks. Can Nightscout help tighten/improve basal rates and bolus insulin to carb ratios by learning over time what works and what doesn't?

**Comment by timomer** [comment]
Do check out the documentation and community for Nightscout , that is your best place to ask that question

**Comment by jamespaulley** [comment]
Will do. Thank you.

**Comment by scottkellie** [comment]
+1 for this feature, just starting out but tight integration with xdrip carbs logged, IOb etc. would be great 👍
